public/caesar-shift/encrypt.py

```diff
@@ -0,0 +1,53 @@
+import string
+from encrypt_library import *
+#START#
+def caesar_shift(plaintext, shift_value=5):
+  ciphertext = ""
+  #<a href="#" class="clickable" class="clickable" onclick="if(enable_tooltips){$('#modalForLoop').modal('open')}">for eachletter in plaintext:</a> 
+  for eachletter in plaintext.lower():
+    #if our letter is not a letter, keep it and move on!
+    if eachletter not in string.ascii_letters:
+      ciphertext = ciphertext + eachletter
+      continue
+
+    #if adding shift constant puts out of alphabet (> ord(z)) then sub ord(a)
+    #<a href="#" class="clickable" onclick="if(enable_tooltips){$('#modalShiftVal').modal('open')}">encoded_letter = shift_by_value_in_alphabet(eachletter, shift_value)</a>
+    encoded_letter = shift_by_value_in_alphabet(eachletter, shift_value)
+
+    #if our shift would go beyond the letter z then we wrap around to the start
+    if is_above_z(encoded_letter):
+      #<a href="#" class="clickable" onclick="if(enable_tooltips){$('#modalWrapA').modal('open')}">encoded_letter = wrap_around_a(encoded_letter)</a>
+      encoded_letter = wrap_around_a(encoded_letter)
+
+    #if our shift would go before the letter a then we wrap around to the end
+    if is_below_a(encoded_letter):
+      #<a href="#" class="clickable" onclick="if(enable_tooltips){$('#modalWrapZ').modal('open')}">encoded_letter = wrap_around_z(encoded_letter)</a>
+      encoded_letter = wrap_around_z(encoded_letter)
+    
+    #put our new letter on the end of our ciphertext
```

Comment:
I would argue string concatenation syntax (addition) might not be intuitive to the students, I want to keep it.